modules/build/src/main/scala/scala/build/options/BuildOptions.scala

```diff
@@ -168,6 +169,55 @@ final case class BuildOptions(
       }
     }
 
+  def latestSupportedScalaVersion(prefixScalaVersion: Option[String]): coursier.core.Version = {
+
+    // used when downloading is fail
+    lazy val defaultStableScalaVersions = StableScalaVersion(
+      version,
+      Seq(defaultScala212Version, defaultScala213Version, defaultScalaVersion)
+    )
+
+    val supportedScalaVersionsUrl = scalaOptions.scalaVersionsUrl
+
+    val task = {
+      val art = Artifact(supportedScalaVersionsUrl).withChanging(true)
+      finalCache.file(art).run.flatMap {
+        case Left(e) => Task.fail(e)
+        case Right(f) => Task.delay {
+            val content = os.read.bytes(os.Path(f, Os.pwd))
+            content
+          }
+      }
+    }
+
+    val launchersTask = finalCache.logger.using(Task.gather.gather(Seq(task)))
+
+    //  If an error occurred while downloading stable versions,
+    //  it uses stable scala versions from Deps.sc
+    val supportedScalaVersions = Try {
+      val scalaVersionJson =
+        launchersTask.unsafeRun()(finalCache.ec).flatten.map(_.toChar).mkString
+
+      upickle.default.read[Seq[StableScalaVersion]](scalaVersionJson)
+    }.getOrElse(Seq(defaultStableScalaVersions))
+
+    val scalaCliVersion = version
+    val stableVersionsForScalaCLI0 = supportedScalaVersions
+      .find(v => scalaCliVersion.contains(v.scalaCliVersion))
+      .map(_.supportedScalaVersions).toSeq.flatten
+
+    val stableScalaVersions =
+      if (stableVersionsForScalaCLI0.isEmpty) defaultStableScalaVersions.supportedScalaVersions
+      else stableVersionsForScalaCLI0
+
+    val prefixScalaVersion0 = prefixScalaVersion.getOrElse("")
+
+    stableScalaVersions
+      .filter(_.startsWith(prefixScalaVersion0))
+      .map(coursier.core.Version(_))
+      .max
```

Comment:
The `.max` may fail if called on an empty seq:
```suggestion
    stableScalaVersions
      .map(coursier.core.Version(_))
      .sorted
      .map(_.repr)
      .reverse
```

Note that this returns a `Seq[String]` now, and that `prefixScalaVersion` is unused.